src/docfx/lib/process/ProcessUtility.cs

```diff
@@ -123,12 +131,12 @@ public static void ReadFile(string path, Action<Stream> read)
     /// Reads the content of a file.
     /// When used together with <see cref="ReadJsonFile{T}(string)"/>, provides inter-process synchronized access to the file.
     /// </summary>
-    public static void WriteJsonFile<T>(string path, T data)
+    public static void WriteJsonFile<T>(string path, T data, bool withIndent = false)
     {
         using (InterProcessMutex.Create(path))
         using (var fs = new FileStream(path, FileMode.Create, FileAccess.Write, FileShare.None, 1024, FileOptions.SequentialScan))
         {
-            JsonSerializer.Serialize(fs, data, s_jsonOptions);
+            JsonSerializer.Serialize(fs, data, withIndent ? s_indentedJsonOptions : s_jsonOptions);
```

Comment:
Can we simply leverage https://github.com/dotnet/docfx/blob/d9d4b7f28d8f0ac0969b3ca59fd63378090821e0/src/docfx/lib/json/JsonUtility.cs#L100-L100?

Reply:
@herohua Good point, I just refactor the code.